hack/calculate-cache.sh

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env bash
+# This script calculates the expected content size, actual cached size, and cache-keys used in caching method prior and after
+# implementation in https://github.com/lima-vm/lima/pull/2508
+#
+# Answer to the question in https://github.com/lima-vm/lima/pull/2508#discussion_r1699798651
+set -u -o pipefail
+
+required_commands=(gh jq limactl sha256sum xxd yq)
+for cmd in "${required_commands[@]}"; do
+	if ! command -v "${cmd}" &>/dev/null; then
+		echo "${cmd} is required. Please install it" >&2
+		exit 1
+	fi
+done
+
+# current workflow uses x86_64 only
+arch=x86_64
+
+LIMA_HOME=$(mktemp -d)
+export LIMA_HOME
+
+# shellcheck disable=SC2034
+macos_12=(
+	# integration
+	examples/default.yaml
+	# vmnet
+	examples/vmnet.yaml
+	# upgrade
+	https://raw.githubusercontent.com/lima-vm/lima/v0.15.1/examples/ubuntu-lts.yaml
+)
+
+# shellcheck disable=SC2034
+ubuntu_2204=(
+	# integration-linux
+	examples/alpine.yaml
+	examples/debian.yaml
+	examples/fedora.yaml
+	examples/archlinux.yaml
+	examples/opensuse.yaml
+	examples/experimental/net-user-v2.yaml
+	examples/experimental/9p.yaml
+	examples/docker.yaml
+	examples/../hack/test-templates/alpine-9p-writable.yaml
+	examples/../hack/test-templates/test-misc.yaml
+)
+
+# shellcheck disable=SC2034
+macos_13=(
+	# vz
+	examples/experimental/vz.yaml
+	examples/fedora.yaml
+)
+
+runners=(
+	macos_12
+	ubuntu_2204
```

Comment:
This is being updated to 24.04
- https://github.com/lima-vm/lima/pull/2542

Reply:
Changed to extract templates from `test.yml` instead of defining shell variables.